substratest/assets.py

```diff
@@ -152,31 +121,93 @@ def load(cls, d):
         mapper = cls.Meta.mapper
         kwargs = {}
         for k, v in d.items():
-            attr_name = mapper[k] if k in mapper else _convert(k)
-            if attr_name not in cls.__annotations__:
-                continue
-            # handle nested dataclasses;
-            # FIXME does not work for list of nested dataclasses
-            attr_type = cls.__annotations__[attr_name]
-            if dataclasses.is_dataclass(attr_type) and isinstance(v, dict):
-                v = attr_type.load(v)
-            kwargs[attr_name] = v
+            if v is not None:
```

Comment:
To minimize nesting, prefer the following syntax:
```python
if v is None:
    continue
```

Minimizing nesting helps code readability.
It also minimizes code changes when modifying the code and avoid having huge diff in git.